RecoHI/HiJetAlgos/plugins/BPMPDInterface.h

```diff
@@ -0,0 +1,848 @@
+					const int index_start = 1)
+		{
+			// Convert from coordinate (COO) storage into compressed
+			// sparse row (CSR)
+
+			std::vector<std::vector<std::pair<int, double> > >
+				column_index(_objective.size(),
+							 std::vector<std::pair<int, double> >());
+
+			std::vector<int>::const_iterator iterator_row =
+				_coefficient_row.begin();
+			std::vector<int>::const_iterator iterator_column =
+				_coefficient_column.begin();
+			std::vector<double>::const_iterator iterator_value =
+				_coefficient_value.begin();
+
+			for (; iterator_value != _coefficient_value.end();
+				 iterator_row++, iterator_column++, iterator_value++) {
+				column_index[*iterator_column].push_back(
+					std::pair<int, double>(
+						// Conversion into Fortran indexing
+						*iterator_row + index_start, *iterator_value));
+			}
+
+			for (std::vector<std::vector<std::pair<int, double> > >::
+					 iterator iterator_outer = column_index.begin();
+				 iterator_outer != column_index.end(); iterator_outer++) {
+				// Conversion into Fortran indexing
+				column_pointer.push_back(row_index.size() + index_start);
+				std::sort(iterator_outer->begin(), iterator_outer->end());
+				for (std::vector<std::pair<int, double> >::const_iterator
+						 iterator_inner = iterator_outer->begin();
+					 iterator_inner != iterator_outer->end();
+					 iterator_inner++) {
+					row_index.push_back(iterator_inner->first);
+					nonzero_value.push_back(iterator_inner->second);
+				}
+			}
+			// Conversion into Fortran indexing
+			column_pointer.push_back(row_index.size() + index_start);
+		}
+	public:
+		problem_t(const bool variable_named)
+			: _optimized(false), _variable_named(variable_named)
+		{
+		}
+		~problem_t(void)
+		{
+			clear();
+		}
+		void clear(void)
+		{
+			clear_row();
+			clear_column();
+			clear_coefficient();
+		}
+		virtual int populate(void) = 0;
+		size_t nrow(void) const
+		{
+			return _rhs.size();
+		}
+		size_t ncolumn(void) const
+		{
+			return _objective.size();
+		}
+		void push_back_row(const char constraint_sense,
+						   const double rhs)
+		{
+			_rhs.push_back(rhs);
+			_constraint_sense.push_back(constraint_sense);
+
+			if (_variable_named) {
+				static const size_t name_length = 24;
+				char *name = new char[name_length];
+
+				snprintf(name, name_length, "c%llu",
+						 static_cast<unsigned long long>(_rhs.size()));
+				_row_name.push_back(name);
+			}
+		}
+		void push_back_row(const std::string &constraint_sense,
+						   const double rhs)
+		{
+			char cplex_sense;
+
+			if (constraint_sense == "<=") {
+				cplex_sense = 'L';
+				push_back_row(rhs, cplex_sense);
+			}
+			else if (constraint_sense == "==") {
+				cplex_sense = 'E';
+				push_back_row(rhs, cplex_sense);
+			}
+			else if (constraint_sense == ">=") {
+				cplex_sense = 'G';
+				push_back_row(rhs, cplex_sense);
+			}
+			else {
+				fprintf(stderr, "%s:%d: illegal sense (`%s')\n",
+						__FILE__, __LINE__, constraint_sense.c_str());
```

Comment:
fprintf(stderr...) is not allowed. Please convert to use the standard LogError().